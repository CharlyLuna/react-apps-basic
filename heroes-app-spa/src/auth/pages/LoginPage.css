
.log-in {
  display: flex;
  flex-direction: column;
  justify-content: center;
  place-items: center;
  margin: 100px auto;
  gap: 1rem;
  max-width: 250px;
}

input {
  width: 100%;
}

button {
  padding-top: 5px;
  width: 100%;
}