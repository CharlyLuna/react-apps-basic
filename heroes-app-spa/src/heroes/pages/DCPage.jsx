import { HeroesList } from '../components'
import { useHeroes } from '../hooks/useHeroes'

export const DCPage = () => {
  const { heroes } = useHeroes({ publisher: 'DC Comics' })
  return (
    <main>
      <h1>DC Comics</h1>
      <HeroesList heroes={heroes} />
    </main>
  )
}
