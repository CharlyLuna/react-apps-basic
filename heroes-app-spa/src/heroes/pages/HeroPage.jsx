import { Navigate, useNavigate, useParams } from 'react-router-dom'
import { getHeroById } from '../helpers/getHeroById'
import { useMemo } from 'react'

export const HeroPage = () => {
  const { id } = useParams()
  const hero = useMemo(() => getHeroById(id), [id])
  const navigate = useNavigate()

  const handleBack = () => {
    navigate(-1)
  }

  if (!hero) {
    return <Navigate to='/heroes/marvel' />
  }

  return (
    <div className='row mt-5'>
      <div className='col-4 animate__animated animate__fadeInLeft'>
        <img className='img-thumbnail' src={`/heroes/${id}.webp`} alt={hero.superhero} />
      </div>
      <div className='col-8 animate__animated animate__bounceInRight'>
        <h3>{hero.superhero}</h3>
        <ul className='list-group list-group-flush'>
          <li className='list-group-item'><b>Alter ego: </b>{hero.alterEgo}</li>
          <li className='list-group-item'><b>Publisher: </b>{hero.publisher}</li>
          <li className='list-group-item'><b>First appearance: </b>{hero.firstAppearance}</li>
        </ul>
        <h4 className='mt-3'>Characters</h4>
        <p>{hero.characters}</p>
        <button className='btn btn-dark' onClick={handleBack}>Back</button>
      </div>
    </div>
  )
}
