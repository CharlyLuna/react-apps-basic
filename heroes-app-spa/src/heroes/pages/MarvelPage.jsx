import { HeroesList } from '../components'
import { useHeroes } from '../hooks/useHeroes'

export const MarvelPage = () => {
  const { heroes } = useHeroes({ publisher: 'Marvel Comics' })

  return (
    <main>
      <h1>Marvel Comics</h1>
      <HeroesList heroes={heroes} />
    </main>
  )
}
