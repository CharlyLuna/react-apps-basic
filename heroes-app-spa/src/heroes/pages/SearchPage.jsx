import { useLocation, useNavigate } from 'react-router-dom'
import { useForm } from '../../hooks/useForm'
import queryString from 'query-string'
import { HeroCard } from '../components/HeroCard'
import { useMemo } from 'react'
import { getHeroesByName } from '../helpers'

export const SearchPage = () => {
  const navigate = useNavigate()
  const location = useLocation()
  const { q = '' } = queryString.parse(location.search)
  const heroes = useMemo(() => getHeroesByName(q), [q])
  const { heroName, onInputChange } = useForm({ heroName: q })

  const showSearch = q.length === 0
  const showError = (q.length > 0) && heroes.length === 0

  const handleSubmit = (e) => {
    e.preventDefault()

    navigate(`?q=${heroName}`)
  }

  return (
    <>
      <h1>Search</h1>
      <div className='row'>
        <div className='col-5'>
          <h4>Search a hero by name</h4>
          <form className='hero-search' onSubmit={handleSubmit}>
            <label htmlFor='heroName' />
            <input
              type='text'
              name='heroName'
              value={heroName}
              autoComplete='off'
              onChange={onInputChange}
            />
            <button className='btn btn-dark mt-2'>Search</button>
          </form>
        </div>
        <div className='col-7'>
          <h4>Results</h4>
          <div
            className='alert alert-primary animate__animated animate__fadeIn'
            style={{ display: showSearch ? '' : 'none' }}
          >
            Search a hero
          </div>
          <div
            className='alert alert-danger animate__animated animate__fadeIn'
            style={{ display: showError ? '' : 'none' }}
          >
            No results found for <b>{q}</b>
          </div>

          {
            heroes?.map(hero => (
              <HeroCard key={hero.id} {...hero} />
            ))
          }
        </div>
      </div>
    </>
  )
}
