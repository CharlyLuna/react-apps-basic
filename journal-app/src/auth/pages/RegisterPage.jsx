import { Button, Grid, Link, TextField, Typography } from '@mui/material'
import { Link as RouterLink } from 'react-router-dom'
import { AuthLayout } from '../layout/AuthLayout'

export const RegisterPage = () => {
  return (
    <AuthLayout title='Register'>
      <form>
        <Grid container>
          <Grid item xs={12} mt={1}>
            <TextField
              label='Full Name'
              type='text'
              placeholder='John Doe'
              fullWidth
            />
          </Grid>
          <Grid item xs={12} mt={1}>
            <TextField
              label='Email'
              type='email'
              placeholder='email@gmail.com'
              fullWidth
            />
          </Grid>
          <Grid item xs={12} mt={1}>
            <TextField
              label='Password'
              type='password'
              placeholder='strongPass#123'
              fullWidth
            />
          </Grid>
          <Grid container spacing={2} sx={{ mb: 2, mt: 1 }}>
            <Grid item xs={12}>
              <Button variant='contained' fullWidth>Create account</Button>
            </Grid>
          </Grid>
          <Grid container direction='row' justifyContent='end'>
            <Typography mr={1}>Already have one?</Typography>
            <Link component={RouterLink} color='inherit' to='/auth/login'>
              Log in
            </Link>
          </Grid>
        </Grid>
      </form>
    </AuthLayout>
  )
}
