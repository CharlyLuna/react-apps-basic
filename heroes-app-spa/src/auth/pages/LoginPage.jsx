import { useNavigate } from 'react-router-dom'
import './LoginPage.css'

export const LoginPage = () => {
  const navigate = useNavigate()

  const handleSubmit = (event) => {
    event.preventDefault()
    navigate('/heroes', {
      replace: true
    })
  }
  return (
    <main>
      <form onSubmit={handleSubmit}>
        <label htmlFor='username'>Username</label>
        <input type='text' name='username' />
        <label htmlFor='password'>Password</label>
        <input type='password' name='password' />
        <button className='btn btn-dark' type='submit'>Log in</button>
      </form>
    </main>
  )
}
