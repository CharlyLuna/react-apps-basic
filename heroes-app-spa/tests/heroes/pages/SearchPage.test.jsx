import { fireEvent, render, screen } from '@testing-library/react'
import { MemoryRouter } from 'react-router-dom'
import { SearchPage } from '../../../src/heroes/pages/SearchPage'
/* eslint-disable no-undef */

const mockedUseNavigate = jest.fn()
jest.mock('react-router-dom', () => ({
  ...jest.requireActual('react-router-dom'),
  useNavigate: () => mockedUseNavigate
}))

describe('Tests in search page', () => {
  beforeEach(() => jest.clearAllMocks())
  test('should show correctly with the default values', () => {
    const { container } = render(
      <MemoryRouter>
        <SearchPage />
      </MemoryRouter>
    )
    expect(container).toMatchSnapshot()
  })
  test('should show Batman card and the input with the querystring value', () => {
    render(
      <MemoryRouter initialEntries={['/search?q=batman']}>
        <SearchPage />
      </MemoryRouter>
    )
    // Check input and image to be what we searched
    const input = screen.getByRole('textbox')
    const img = screen.getByRole('img')
    expect(input.value).toBe('batman')
    expect(img.src).toContain('/heroes/dc-batman.webp')
    // Check that the not found message is not appearing
    const errorMessage = screen.getByTestId('error-message')
    expect(errorMessage.style.display).toBe('none')
  })
  test('should show an error if the hero is not found', () => {
    render(
      <MemoryRouter initialEntries={['/search?q=b231']}>
        <SearchPage />
      </MemoryRouter>
    )
    const errorMessage = screen.getByTestId('error-message')
    expect(errorMessage.style.display).toBe('')
    expect(errorMessage.textContent).toBe('No results found for b231')
  })
  test('should call navigate to the new screen', () => {
    render(
      <MemoryRouter initialEntries={['/search']}>
        <SearchPage />
      </MemoryRouter>
    )
    const input = screen.getByRole('textbox')
    const button = screen.getByRole('button')
    fireEvent.change(input, { target: { value: 'batman' } })
    fireEvent.click(button)
    expect(mockedUseNavigate).toHaveBeenCalledWith('?q=batman')
  })
})
