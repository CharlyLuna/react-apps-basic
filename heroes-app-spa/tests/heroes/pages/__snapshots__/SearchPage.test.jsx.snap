// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests in search page should show correctly with the default values 1`] = `
<div>
  <h1>
    Search
  </h1>
  <div
    class="row"
  >
    <div
      class="col-md-5"
    >
      <h4>
        Search a hero by name
      </h4>
      <form
        class="hero-search"
      >
        <label
          for="heroName"
        />
        <input
          autocomplete="off"
          name="heroName"
          type="text"
          value=""
        />
        <button
          class="btn btn-dark mt-2"
        >
          Search
        </button>
      </form>
    </div>
    <div
      class="col-md-7 mt-4 mt-md-0"
    >
      <h4>
        Results
      </h4>
      <div
        class="alert alert-primary animate__animated animate__fadeIn"
      >
        Search a hero
      </div>
      <div
        class="alert alert-danger animate__animated animate__fadeIn"
        data-testid="error-message"
        style="display: none;"
      >
        No results found for 
        <b />
      </div>
    </div>
  </div>
</div>
`;
